flatgeobuf/index.qmd

```diff
@@ -0,0 +1,95 @@
+
+A compression like gzip can be applied to the FlatGeobuf file in full, but keep in mind that storing the compressed file will eliminate random access support.
+
+### No append support
+
+FlatGeobuf is a write-only format, and doesn't support appending, as that would invalidate the spatial index in the file.
+
+### Random access supported via spatial index
+
+FlatGeobuf optionally supports a spatial index at the beginning of the file, which can speed up reading portions of a file based on a spatial query. For more information on how this spatial index works, refer to the [Hilbert R Tree](./hilbert-r-tree.qmd) page.
+
+<!--
+TODO: Not sure whether this section makes sense to include
+
+### No internal chunking
+
+FlatGeobuf has no internal chunking. Rather, the binary features are stored sequentially with no other grouping.
+
+One implication of this is that you have
+
+And because FlatGeobuf is a row-based format, if some properties are really large, like if a column is a string description with many characters, you'll have a tradeoff between fetching lots of extra data to combine multiple requests into one or making many requests for small features that are scattered throughout the file. -->
+
+### Streaming features is supported
+
+FlatGeobuf supports _streaming_, meaning that you can use part of the file before the entire file has finished downloading. This is different than random access, because you have no ability to skip around in the file.
+
+Streaming can be valuable because it makes an application seem more responsive; you can have _something_ happen without having to wait for the full download to complete. A good example of this is [this example](https://observablehq.com/@bjornharrtell/streaming-flatgeobuf) by FlatGeobuf's author Björn Harrtell. As the file is downloaded to the browser, portions of the file get rendered progressively in parts.
+
+This works even with full-file compression like gzip or deflate because those compression algorithms support streaming decompression.
+
+### Broad type system
+
+FlatGeobuf supports attributes with a range of types:
+
+- `Byte`: Signed 8-bit integer
+- `UByte`: Unsigned 8-bit integer
+- `Bool`: Boolean
+- `Short`: Signed 16-bit integer
+- `UShort`: Unsigned 16-bit integer
+- `Int`: Signed 32-bit integer
+- `UInt`: Unsigned 32-bit integer
+- `Long`: Signed 64-bit integer
+- `ULong`: Unsigned 64-bit integer
+- `Float`: Single precision floating point number
+- `Double`: Double precision floating point number
+- `String`: UTF8 string
+- `Json`: General JSON type intended to be application specific
+- `DateTime`: ISO 8601 date time
+- `Binary`: General binary type intended to be application specific
+
+Note that FlatGeobuf is unable to store nested types without overhead. It doesn't support a "list" or "dict" type apart from JSON, which has a parsing overhead.
```

Comment:
This feels a little out of place, but perhaps an example would help. Could we make it a callout box?

```suggestion
::: {.callout-note}
Note that FlatGeobuf is unable to store nested types without overhead. It doesn't support a "list" or "dict" type apart from JSON, which has a parsing overhead.

In some situations, having strong nested type support can be useful. For example STAC stored as GeoParquet has columns that are nested, such as the `assets` column that needs to store a dictionary-like mapping from asset names to their information. FlatGeobuf is able to store such data by serializing it to JSON, but it's not possible to see the nested schema before parsing the full dataset.
:::
```